sketchcrapp.sh

```diff
@@ -130,120 +135,187 @@ clean() {
 # and link-resolve the patched Sketch.app
-# todo: *
-path() {
-  echo "[+] Selected Sketch.app version is $version ... SketchCrapp starting ... OK"
-  echo "[+] Selected Sketch.app path is </Applications> (auto-detected) ... OK"
-  echo "[+] Detecting current binary hash (MD5) ... OK"
-  echo "[+] Patching offset for $version ..."
-  # todo: offset_patch ($offset, $instruction)
-  
-  echo "[+] Generating self-signed certificate ..."
-  # genSelfCert()
-  # importSelfSignCert()
-  # signApplication()
-  # cleanup()
-  
+engin() {
+  appVersion="$1"
+  appPath="$2"
+  execPath="$3"
+  #Version Selector
+  echo "[+] Selected Sketch.app version is $appVersion ... SketchCrapp starting ... OK"
+  echo "[+] Patching offset for $appVersion ..."
+  case "$appVersion" in
+    "63.1")
+      patch "${address_param_631[*]}" "${value_param_631[*]}" "$execPath"
+      ;;
+    "64")
+      echo "select 64"
+      ;;
+    "65.1")
+      echo "select 65.1"
+      ;;
+    "66.1")
+      echo "select 66.1"
+      ;;
+    "67.1")
+      echo "select 67.1"
+      ;;
+    "67.2")
+      echo "select 67.2 but not support yet."
+      ;;  
+    *)
+      echo "not support."
+  esac
+  # CodeSigning area
+  # check if sketchcrapp certificate already exist.
+  if ! security find-certificate -c "sketchcrapp" 2>&1 >/dev/null; then
+    # certificate not exist, generate one.
+    genSelfSignCert
+    # import the certificate.
+    importSelfSignCert
+  else 
+    echo "sketchcrapp certificate already exist. using exist one."
+  fi
+  # Sign the application.
+  signApplication "$appPath"
+  # call cleaner to do some housekeeping.
+  clean
   echo "[+] SketchCrapp process completed. Sketch.app has been patched :)"
-  # todo: replace <hash> with output of md5sum
-  echo "[+] Binary Hash (before): <hash> [MD5]"
-  echo "[+] Binary Hash  (after): <hash> [MD5]"
-
+  echo "[+] If dialogue show up said “Sketch 3.app” can’t be opened "
+  echo "[+] because Apple cannot check it for malicious software."
+  echo "[+] Please right-click the application and select open."
   echo ""
   echo "SketchCrapp (A Sketch.app cracking tool)"
   echo "https://github.com/duraki/SketchCrapp"
 }
-
-offset_patch() {
-  # todo: do binary patching here
-  # echo " > now hacking $offset > $offset instruction ... <OK>"
-}
-
-# Command Line Interface initialization.
-## > Missing command arg: Version
-if [ $# -eq 0 ]; then
-  echo "SketchCrapp requires Sketch version as an input value."
-  echo "Use -h for more information."
-  exit 0
-fi
-
-## > Missing openssl library
+## > Check if missing openssl library
 if ! command -v openssl &> /dev/null; then
-  echo "OpenSSL is not installed on your system. This should not happen, macOS have openssl built-in."
+  echo "OpenSSL is not installed on your system."
+  echo "This should not happen, macOS have openssl built-in."
   echo "[FIX] Try: brew install openssl"
+  echo "[FIX] Try: port install openssl"
   echo "[FIX] Try: install openssl manually"
```

Comment:
We should exit() here.

Reply:
finish in 4fcadbea796473356c258aada8608c31f59fdab8